tests/unit/clients/test_http.py

```diff
@@ -245,6 +245,36 @@ def client(mocker, auth_client):
     assert 2 == len(caplog.records)
 
 
+@pytest.mark.asyncio
+async def test_request_with_other_exception(
+        client, mocker, monkeypatch, caplog):
+    """Correct exception is raised when the aiohttp request itself fails."""
+
+    async def mock_refresh_token():
+        pass
+
+    monkeypatch.setattr(
+        client._auth_client, 'refresh_token', mock_refresh_token)
+
+    async def mock_valid_token_set():
+        pass
+
+    monkeypatch.setattr(client, 'valid_token_set', mock_valid_token_set)
+
+    mock_aiohttp_request = mocker.Mock()
+    mock_aiohttp_request.side_effect = ValueError  # randomly chosen
```

Comment:
Doesn't it need to be instantiated? i.e `ValueError('not a valid value')`?

Reply:
Exceptions default to ('') iirc.  In any case, it works.  But I can add that to further test the log message.

Reply:
Oh wait, you mean because the 'side effect' is a class.  The mock library checks if it's an exception and instantiates it for you.